packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,127 @@
+
+  @override
+  State<NeonUserAvatar> createState() => _UserAvatarState();
+}
+
+class _UserAvatarState extends State<NeonUserAvatar> {
+  late final _userStatusBloc = Provider.of<AccountsBloc>(context, listen: false).getUserStatusesBloc(widget.account);
+
+  @override
+  void initState() {
+    super.initState();
+
+    unawaited(_userStatusBloc.load(widget.username));
+  }
+
+  @override
+  Widget build(final BuildContext context) {
+    final isDark = Theme.of(context).brightness == Brightness.dark;
+    final size = (widget.size * MediaQuery.of(context).devicePixelRatio).toInt();
+    return Stack(
+      alignment: Alignment.center,
+      children: [
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-${widget.username}-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: widget.username,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: widget.username,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        if (widget.showStatus) ...[
+          ResultBuilder<NextcloudUserStatusPublicStatus?>(
+            stream: _userStatusBloc.statuses.map((final statuses) => statuses[widget.username]),
+            builder: _userStatusIconBuilder,
+          ),
+        ],
+      ],
+    );
+  }
+
+  Widget _userStatusIconBuilder(final BuildContext context, final Result<NextcloudUserStatusPublicStatus?> result) {
+    final hasEmoji = result.data?.icon != null;
+    final scaledSize = widget.size / (hasEmoji ? 2 : 3);
+
+    Widget? child;
+    Decoration? decoration;
+    if (result.loading) {
+      child = CircularProgressIndicator(
+        strokeWidth: 1.5,
+        color: Theme.of(context).colorScheme.onPrimary,
+      );
+    } else if (result.error != null) {
+      child = Icon(
+        Icons.error_outline,
+        size: scaledSize,
+        color: Colors.red,
+      );
+    } else if (hasEmoji) {
+      child = Text(
+        result.data!.icon!,
+        style: const TextStyle(
+          fontSize: 16,
+        ),
+      );
+    } else if (result.data != null) {
+      decoration = BoxDecoration(
+        shape: BoxShape.circle,
+        color: _userStatusToColor(result.data!.status),
+      );
+    }
+
+    return SizedBox.square(
+      dimension: widget.size,
+      child: Align(
+        alignment: Alignment.bottomRight,
+        child: SizedBox.square(
+          dimension: scaledSize,
+          child: DecoratedBox(
+            decoration: decoration ?? const BoxDecoration(),
+            child: child,
+          ),
+        ),
```

Comment:
can we make this:

```dart
        child: Container(
          width: scaledSize,
          height: scaledSize,
          decoration: decoration,
          child: child,
        ),
```

We lose the square information but it shows that this is one container that's aligned to the bottom and has a decoration (also the decoration can be null avoiding confusion about that).

I'm not 100% sure if it improves readability or not xD